doc/system_specific_setup.rst

```diff
@@ -1,225 +1,266 @@
-added to the PYTHONPATH, e.g.:
-::
+        > pip install --user fparser pyparsing
 
-   > export PYTHONPATH=/home/a_user/.local/lib/python2.7/site-packages:${PYTHONPATH}
+    If necessary, add the ``--user`` option to all ``pip`` commands and
+    remove all ``sudo`` commands.
 
-Alternatively, if pip is not an option, a tarball of the latest release
-may be downloaded from https://github.com/stfc/fparser/releases. Simply
-unpack the tarball and ensure that the resulting
-``fparser-x.y.z/src/fparser`` directory is in your PYTHONPATH.
+    Uninstalling is simply a matter of doing::
+
+       > sudo pip uninstall fparser pyparsing
 
 PSyclone supports the ability to output a schedule dependency graph
 using the graphviz package. This is optional and the associated
 routine will silently return if the graphviz bindings are not
-installed. To output a graph you need to install the graphviz package
-::
-
-   > sudo apt-get install graphviz
-
-and the Python bindings to the graphviz package
+installed. If you have the graphviz package installed (see
+especially section :ref:`ubuntu_user` if you are on Ubuntu), you also need
+to install the python bindings to the graphviz package:
 ::
 
    > sudo pip install graphviz
 
 If you just want to use PSyclone then you've installed all you need
-and you're ready to go to the getting-going :ref:`getting-going-run`
-section.
+and you are ready to go to the getting-going :ref:`getting-going-run` section.
+
+.. _dev_setup:
 
 Developer set-up
-++++++++++++++++
+----------------
 
 This section adds software that is used to develop and test
-PSyclone. Note, we assume you have already installed the software
-described in the :ref:`users` section.
+PSyclone. It includes all packages for testing and creation of
+documentation in html and pdf. We assume you have already installed the software
+described in the :ref:`setup_user` section.
 
-pytest
-^^^^^^
+.. _psyclone_from_git:
 
-Install pytest for running python tests
-::
+Installing PSyclone From Git
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+For development it is recommended to get a copy of PSyclone using git to get 
+access to the latest development version.
 
-   > sudo apt-get install python-pytest
+Installing git for Ubuntu
++++++++++++++++++++++++++
+You need to install the git package::
 
-You can now run the PSyclone python tests:
+    > sudo apg-get install git
 
-.. parsed-literal::
 
-   > cd PSyclone-\ |release|\ /src/psyclone/tests
-   > py.test
+Installing git on OpenSUSE
+++++++++++++++++++++++++++
+You need to install the git package::
+
+    >> sudo zypper --no-recommends install git
 
-Documentation
-^^^^^^^^^^^^^
 
+Cloning PSyclone using git
+++++++++++++++++++++++++++
+Cloning PSyclone from git and setting up your environment is done as follows::
+
+   > cd <PSYCLONEHOME>
+   > git clone https://github.com/stfc/PSyclone.git
+   > cd PSyclone
+   > export PYTHONPATH=`pwd`/src:${PYTHONPATH}
+   > export PATH=`pwd`/bin:${PATH}
```

Comment:
I recommend we replace the two "export" lines with "pip install --user -e ." with a note to say that the "-e" flag causes the project to be installed in 'editable' mode so that any changes to the PSyclone source take effect immediately.

Reply:
We can then drop the text below about setting your PYTHONPATH permanently.

Reply:
I've done that - but I can't see how this sets PATH for the user?

Reply:
I guess it comes back to whether or not your linux distribution sets things up for you. On Fedora it automatically puts ~/.local/bin on your PATH and ~/.local/lib/python... on your PYTHONPATH so the user doesn't have to do anything else after the `pip install`.

Reply:
Ah, thanks - I never thought of looking in ~/.local/bin ... since it is indeed not part of your path on OpenSUSE. I added this to my latest commit.